backend/fms_core/schema_validators.py

```diff
@@ -107,20 +107,21 @@ def deconstruct(self):
     "properties": {
         "run": {"type": "string"},
         "lane": {"type": "string", "pattern": str(r"^([1-9][0-9]*|0)$")},
-        "run_obj_id": {"type": ["null", "number"]},
+        "run_obj_id": {"type": "number"},
         "metrics_report_url": {"type": "string"},
         "readsets": {
             "type": "object",
             "patternProperties": {
                 "^.*$": {
                     "type": "object",
                     "properties": {
+                        "project_obj_id": {"type": "string"},
                         "external_project_id": {"type": "string"},
                         "project_name": {"type": "string"},
                         "sample_name": {"type": "string"},
                         "derived_sample_id": {"type": ["null", "number"]},
                     },
-                    "required": ["sample_name", "external_project_id", "project_name"]
+                    "required": ["project_obj_id", "sample_name", "external_project_id", "project_name"]
```

Comment:
I guess we are still handling `external_project_id`?

Reply:
It is still the unifying ID for project in the center ... if the json ever get ingested elsewhere it would still be useful ... maybe i could remove it as a requirement but it might be better to keep the pressure for compliance on that side.